service/inbox.tsp

```diff
@@ -0,0 +1,102 @@
+import "@typespec/http";
+import "@typespec/openapi";
+
+using Http;
+
+@tag("Inbox")
+namespace CoreSystem.Inbox {
+  @doc("The structure of a user inbox message with customized flag.")
+  model UserInboxMessage {
+    @doc("The userId or the orgId of the inbox owner.")
+    id: uuid;
+
+    @doc("The inbox message this user inbox message refers to.")
+    message: InboxMessage;
+
+    @doc("The type of the user inbox message.")
+    type: UserInboxMessageFilters;
+  }
+
+  @doc("The filters that can be applied by the user for their inbox messages.")
+  model UserInboxMessageFilters {
+    @doc("Indicates whether the user has read the inbox message.")
+    isRead: boolean = false;
+
+    @doc("User can mark the inbox message as starred.")
+    isStarred: boolean = false;
+
+    @doc("User can archive the inbox message and hide it from the inbox view.")
+    isArchived: boolean = false;
+  }
+
+  alias contentType = "text" | "form";
+
+  @doc("The structure of a full inbox message.")
+  model InboxMessage {
+    @doc("The unique identifier of an inbox message.")
+    id: uuid;
+
+    @doc("Sender may be the org-{orgId} or unit-{unitId}.")
+    postedBy: uuid;
+
+    @doc("The title of the inbox message.")
+    title: string;
+
+    @doc("The subtitle of the inbox message.")
+    subtitle: string;
+
+    @doc("To decide whether to reference the content from Text API or Forms API.")
+    type: contentType;
+
+    @doc("(Optional) The identifier of the text content in markdown.")
+    textId?: uuid;
+
+    @doc("(Optional) The identifier of the form content.")
+    formId?: uuid;
+
+    @doc("(Optional) The due date of the inbox message.")
+    dueDate?: utcDateTime;
+
+    @doc("The post timestamp of the inbox message.")
+    createdAt: utcDateTime;
+
+    @doc("The last updated timestamp of the inbox message.")
+    updatedAt: utcDateTime;
+  }
+
+  @doc("Get all the preview of the inbox messages with pagination.")
+  @route("/inbox")
+  @get
+  @list
+  op getInboxMessages(
+    @query filters?: UserInboxMessageFilters,
```

Comment:
I think it’s fine to write

```
@query isRead?: boolean,
@query isStarred?: boolean,
@query isArchived?: boolean,
```

directly. Otherwise, using

```
@query filters?: UserInboxMessageFilters
```

still results in an incorrect OpenAPI YAML format. More specifically, it will generate

```yaml
- name: filters
  in: query
  required: false
  schema:
    $ref: '#/components/schemas/Inbox.UserInboxMessageFilters'
  explode: false
```

for the filter, but our intention is not to put JSON directly in the URL query.